va/win32/va_win32.h

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright © Microsoft Corporation
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice (including the next
+ * paragraph) shall be included in all copies or substantial portions of the
+ * Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ */
+
+#ifndef _VA_WIN32_H_
+#define _VA_WIN32_H_
+
+#include <va/va.h>
+#include <windows.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** \brief HANDLE memory type
+ * 
+ * The HANDLE type can be used with D3D12 sharing resource functions such as:
+ * https://docs.microsoft.com/en-us/windows/win32/api/d3d12/nf-d3d12-id3d12device-opensharedhandle
+ * https://docs.microsoft.com/en-us/windows/win32/api/d3d12/nf-d3d12-id3d12device-createsharedhandle
+ *
+ * When used with VASurfaceAttribExternalBufferDescriptor for vaCreateSurfaces(...,num_surfaces,...),
+ * the associated VAGenericValueTypePointer is of type HANDLE[num_surfaces]
```

Comment:
I believe that other kinds of external surfaces follow the pattern that there is single handle in the attribute and functions should be called as `vaCreateSurfaces(...,1,...)`. Do we actually need to step out from this pattern for Windows?

Reply:
I think this is true for `VA_SURFACE_ATTRIB_MEM_TYPE_DRM_PRIME_2`, however for `VA_SURFACE_ATTRIB_MEM_TYPE_DRM_PRIME` when the passed structure is `VASurfaceAttribExternalBuffers`, there are two params `buffers` and `num_buffers` for passing individual handles for each of the `num_surfaces` to be created.

I found two driver implementations that implement vaCreateSurfaces**2** like that (not to confuse with the older vaCreateSurfaces function signature that doesn't receive the attribs as params). 

- https://github.com/intel/media-driver/blob/1faefc4110cb701b88a4942804b31ab19f3e1a3a/media_driver/linux/common/ddi/media_libva.cpp#L2918
    - Which I think ends up opening the each handle in https://github.com/intel/media-driver/blob/69235cbdfa375aefeded9afe1cc86b826ebbcf55/media_driver/linux/common/ddi/media_libva_util.cpp#L350

- https://gitlab.freedesktop.org/mesa/mesa/-/blob/main/src/gallium/frontends/va/surface.c#L1024

I think it's useful to be able to import N handles at once into N created surfaces in Windows following the existing pattern for `VA_SURFACE_ATTRIB_MEM_TYPE_DRM_PRIME`.

Reply:
Personally I believe that having the `VA_SURFACE_ATTRIB_MEM_TYPE_` spread out across headers is a bad idea. Although I also think that we should adapt the vaGetDisplay interface after the EGL_PLATFORM extensions, since basically that's what the libva design was after.

Regardless - these are orthogonal to the feature added here and can be looked into at a later date.

Reply:
Ok, till that's documented (and it is for win surfaces) I am fine with that. Leave comment unresolved, please, for other people to provide feedback.